rules/rbe_repo.bzl

```diff
@@ -0,0 +1,560 @@
+                      bazel_rc_version)
+    else:
+        bazel_url += "/release/bazel-" + bazel_version
+    bazel_url += "-installer-linux-x86_64.sh"
+    install_bazel_cmd = ["bazel_url=" + bazel_url]
+    install_bazel_cmd += ["mkdir -p /src/bazel"]
+    install_bazel_cmd += ["cd /src/bazel/"]
+    install_bazel_cmd += ["wget $bazel_url --no-verbose --ca-certificate=/etc/ssl/certs/ca-certificates.crt -O /tmp/bazel-installer.sh"]
+    install_bazel_cmd += ["chmod +x /tmp/bazel-installer.sh"]
+    install_bazel_cmd += ["/tmp/bazel-installer.sh"]
+    install_bazel_cmd += ["rm -f /tmp/bazel-installer.sh"]
+
+    # Command to recursively convert soft links to hard links in the config_repos
+    deref_symlinks_cmd = []
+    for config_repo in _CONFIG_REPOS:
+        symlinks_cmd = ("find $(bazel info output_base)/" +
+                        _EXTERNAL_FOLDER_PREFIX + config_repo +
+                        " -type l -exec bash -c 'ln -f \"$(readlink -m \"$0\")\" \"$0\"' {} \;")
+        deref_symlinks_cmd.append(symlinks_cmd)
+    deref_symlinks_cmd = " && ".join(deref_symlinks_cmd)
+
+    # Command to copy produced toolchain configs to a tar at the root
+    # of the container.
+    copy_cmd = ["mkdir " + _OUTPUT_DIR]
+    for config_repo in _CONFIG_REPOS:
+        src_dir = "$(bazel info output_base)/" + _EXTERNAL_FOLDER_PREFIX + config_repo
+        copy_cmd.append("cp -dr " + src_dir + " " + _OUTPUT_DIR)
+    copy_cmd.append("tar -cf /" + outputs_tar + " -C " + _OUTPUT_DIR + "/ . ")
+    output_copy_cmd = " && ".join(copy_cmd)
+
+    # if use_default_project was selected, we need to modify the WORKSPACE and BUILD file
+    setup_default_project_cmd = ["cd ."]
+    if use_default_project:
+        setup_default_project_cmd += ["cd " + _BAZEL_CONFIG_DIR + "/" + _PROJECT_REPO_DIR]
+        setup_default_project_cmd += ["mv BUILD.sample BUILD"]
+        setup_default_project_cmd += ["touch WORKSPACE"]
+
+    bazel_cmd = "cd " + _BAZEL_CONFIG_DIR + "/" + _PROJECT_REPO_DIR
+
+    # For each config repo we run the target @<config_repo>//...
+    bazel_targets = "@" + "//... @".join(_CONFIG_REPOS) + "//..."
+    bazel_flags = ""
+    if not ctx.attr.incompatible_changes_off:
+        bazel_flags += " --all_incompatible_changes"
+    bazel_cmd += " && bazel build " + bazel_flags + " " + bazel_targets
+
+    # Command to run to clean up after autoconfiguration.
+    # we start with "cd ." to make sure in case of failure everything after the
+    # ";" will be executed
+    clean_cmd = "cd . ; bazel clean"
+    if use_default_project:
+        clean_cmd += "; rm WORKSPACE ; mv BUILD BUILD.sample"
+
+    docker_cmd = [
+        "#!/bin/bash",
+        "echo === Starting docker autoconfig ===",
+        ctx.attr.setup_cmd,
+    ]
+    docker_cmd += install_bazel_cmd
+    docker_cmd += setup_default_project_cmd
+    docker_cmd += [
+        bazel_cmd,
+        deref_symlinks_cmd,
+        output_copy_cmd,
+        clean_cmd,
+    ]
```

Comment:
why not
    docker_cmd = [
        "#!/bin/bash",
        "echo === Starting docker autoconfig ===",
        ctx.attr.setup_cmd,
        install_bazel_cmd,
        setup_default_project_cmd,
        bazel_cmd,
        deref_symlinks_cmd,
        output_copy_cmd,
        clean_cmd,
    ]